users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -401,6 +616,203 @@ def process_args(args: Dict[str, Any]):
+        {"speed": {"prob": 0.6, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.6, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 1, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 1, "minimum": 0.8, "maximum": 1.2}},
+        {"non_linearity": {"prob": 0.7, "minimum": 0.9, "maximum": 1.1}},
+        {"non_linearity": {"prob": 0.3, "minimum": 0.9, "maximum": 1.1}},
+        {"non_linearity": {"prob": 1, "minimum": 0.9, "maximum": 1.1}},
+        # v2
+        {"tempo": {"prob": 0.8, "minimum": 0.65, "maximum": 1.35}},
+        {
+            "tempo": {"prob": 0.8, "minimum": 0.7, "maximum": 1.3},
+            "non_linearity": {"prob": 0.3, "minimum": 0.9, "maximum": 1.1},
+        },
+        {
+            "tempo": {"prob": 0.3, "minimum": 0.7, "maximum": 1.3},
+            "non_linearity": {"prob": 0.3, "minimum": 0.9, "maximum": 1.1},
+        },
+        {"non_linearity": {"prob": 0.3, "minimum": 0.85, "maximum": 1.15}},
+    ]
+
+    # seperated because the training needs a different network without preemphasis
+    perturbation_args_preemphasis = [
+        {"preemphasis": {"prob": 0.3, "minimum": 0.94, "maximum": 1.0, "default": 0.97}},
+        {"preemphasis": {"prob": 0.7, "minimum": 0.94, "maximum": 1.0, "default": 0.97}},
+        {"preemphasis": {"prob": 1.0, "minimum": 0.94, "maximum": 1.0, "default": 0.97}},
+    ]
+
+    def process_args(args: Dict[str, Any]):
+        """
+        Process the argument dictionary to generate a key string and a report string.
+
+        Returns:
+            tuple: A tuple containing the key string and the report string.
+        """
+
+        key_string = ""
+        report_dict = {}
+
+        for key, value in args.items():
+
+            if key in ["speed", "tempo", "preemphasis", "non_linearity"]:
+                key_component = f"{key}_{value['prob']}_{value['minimum']}_{value['maximum']}"
+                if "default" in value:
+                    key_component += f"_{value['default']}"
+                key_string += key_component
+                report_dict[key] = f"{value['prob']}_{value['minimum']}_{value['maximum']}"
+            elif key == "codecs":
+                codecs_str = "_".join(
+                    [f"{codec['encoding']}_{codec['prob']}_{codec['minimum']}_{codec['maximum']}" for codec in value]
+                )
+                key_string += f"{key}_{codecs_str}_"
+                report_dict[key] = codecs_str
+            else:
+                raise ValueError(f"Unknown argument name: {key}")
+
+        return key_string, report_dict
+
+    nn_base_args = {}
+
+    for args in perturbation_args:
+        exp_name_suffix, report_dict = process_args(args)
+
+        # Construct the exp_name and report_args
+        exp_name = f"scf_bs2x5k_perturb_from_checkpoint_24_{exp_name_suffix}"
+        report_args = report_dict
+        nn_base_args[exp_name] = dict(
+            returnn_args={
+                "extra_args": {
```

Comment:
Couldn't all of this be moved to the initial definition of `returnn_args`?

Reply:
Most of the extra args like the checkpoint and the `audio_perturb_args` are not shared and need to be there.

Reply:
Right, `audio_perturb_args` changes. The rest can then stay even though it is static.